econml/shap.py

```diff
@@ -113,7 +113,13 @@ def _shap_explain_model_cate(cme_model, models, X, d_t, d_y, feature_names=None,
         each treatment name (e.g. "T0" when `treatment_names=None`) as key
         and the shap_values explanation object as value.
     """
-
+    def fall_back():
+        return _shap_explain_cme(cme_model, X, d_t, d_y,
+                                 feature_names=feature_names,
+                                 treatment_names=treatment_names,
+                                 output_names=output_names,
+                                 input_names=input_names,
+                                 background_samples=background_samples)
```

Comment:
Why you do it in this way? when this function is called, it will still pass in the updated feature_names/output_names/feature_names, although it's fine to run `_define_names` function twice here. Then no need to define this `fall_back` function repeatedly under each function?

Reply:
I dont think so. These variables I believe are bound at the time of the creation of the function, so that would ensure that these are the raw input cariables

Reply:
You're right. Let me change that. Good catch! Was mistaken about lazy/eager evaluation here. I still want to avoid define_names being called twice, but I thought this was a short cut

Reply:
Made the change. Let me know if that is ok.

Reply:
Looks great!

Reply:
sweet! Can you also approve the PR?